### PR TITLE
feat: adjust default config path while preserving backwards compatibility

### PR DESCRIPTION
**Pull request by MalteJanz**
## What changed?

Changes the path where config files are stored for new projects / extensions while keeping them in their original location for existing projects / extensions.

## Why?

For context [read the ADR](https://github.com/shopware/shopware-cli/blob/main/docs/adr/0002-project-config-dot-config.md), TLDR: we want to adopt [the `.config` convention](https://dot-config.github.io/)

## How was this tested?

Besides the changed / added automated tests:
- run `shopware-cli project config init` and make sure it's stored in the new / recommended location
- run `shopware-cli project doctor` or any other command that uses the config, make sure it is picked up
- try different filenames / locations to make sure the correct one is loaded according to priority (if multiple exists): `.config/shopware-project.yml` > `.shopware-project.yaml` > `.shopware-project.yml`
  - as well as a warning is displayed that multiple exists
- perform similar steps for `shopware-cli extension config init` and related commands

## Related issue or discussion

part of https://github.com/shopware/shopware-cli/issues/1387

## Whats still missing before this is ready

- [x] make similar adjustments for the implementations of `shopware-extension.yml`, right now only `.shopware-project.yml` is covered
- [ ] do a Demo with our QA to make sure no edge cases are overlooked
- [x] add further automated tests, ideally raising the coverage
- [x] wait until a similar adjustment is ready in [deployment-helper](https://github.com/shopware/deployment-helper) so it can be merged + released together
  - related deployment-helper change: https://github.com/shopware/deployment-helper/pull/114 

## Summary by CodeRabbit

- **New Features**
  - Project and extension configurations now default to `.config/` locations.
  - Legacy filenames and extensions remain supported through automatic discovery and priority-based selection.
  - Configuration updates preserve the loaded file’s original location.
  - Initialization commands prevent accidental overwrites and display the resolved configuration path.
  - Deployment archives recognize the new project configuration location.

- **Documentation**
  - Updated CLI guidance, user-facing messages, and documentation to reflect supported locations and discovery order.

### CODE REVIEW

File: cmd/project/project_config_init.go [hunk 1]
```diff
@@ -14,8 +14,8 @@ import (
 
 var projectConfigInitCmd = &cobra.Command{
 	Use:   "init",
-	Short: "Creates a new project config in current dir",
-	Long: `Creates a new .shopware-project.yml in the current directory.
+	Short: "Creates a new project config",
+	Long: `Creates a new config in the current directory under .config/shopware-project.yaml .
```
Comment (MalteJanz) [inline cmd/project/project_config_init.go:18]: **Reminder:** do not merge until a PR in [deployment-helper](https://github.com/shopware/deployment-helper) is ready, as that relying on the same config file:
https://github.com/shopware/deployment-helper/blob/151383a0b1dafae673c8bafe5dea5b80816a8267/src/Config/ConfigFactory.php#L22

File: internal/extension/config_init.go [hunk 2]
```diff
@@ -9,15 +9,16 @@ import (
...
-
 	// ConfigSchemaURL is the JSON Schema URL used by the YAML language server.
 	ConfigSchemaURL = "https://shopware.github.io/shopware-cli/shopware-extension-schema.json"
 )
 
+var ConfigLocations = []string{
+	".config/shopware-extension.yml", // recommended location
+	".shopware-extension.yml",
+	".shopware-extension.yaml",
+}
```
Comment (MalteJanz) [inline internal/extension/config_init.go:22]: Todo: consider issuing a warning if multiple exists, similar to the one in https://github.com/shopware/shopware-cli/blob/791ecedbf70663c25d16d56bd6b232164c262629/internal/shop/config.go#L995

**Review by shyim — APPROVED** [review]
lgtm. nice with the warning when old and new config is there

File: internal/extension/config_init.go [hunk 3]
```diff
@@ -1,23 +1,26 @@
...
-	ConfigFileNameAlt = ".shopware-extension.yaml"
-
 	// ConfigSchemaURL is the JSON Schema URL used by the YAML language server.
 	ConfigSchemaURL = "https://shopware.github.io/shopware-cli/shopware-extension-schema.json"
 )
 
+var ConfigLocations = []string{
+	".config/shopware-extension.yml", // recommended location
+	".shopware-extension.yml",
+	".shopware-extension.yaml",
```
Comment (MalteJanz) [inline internal/extension/config_init.go:21]: Copilot raised an interesting point in my docs PR:
https://github.com/shopware/docs/pull/2521#discussion_r4035207357

Because this is part of our docs right now:

> Because `.config/shopware-extension.yml` is not included in the extension ZIP created by Shopware CLI, keeping Store metadata in the repository does not add the listing configuration to the distributed package.


Looks like neither right now nor with this PR these config files are stripped from the packaged extension:
https://github.com/shopware/shopware-cli/blob/f9806a41ea74aca3e9432709fd8dedd5b9e528c1/internal/extension/cleanup.go#L12-L74

Should I change that? The `.local` override could contain sensitive information right? At least the project one, not sure about the extension one.

Any opinions on this @shyim / @Ant1gua / @tturkowski ?

File: cmd/project/project_deployment_package.go [hunk 4]
```diff
@@ -23,5 +21,5 @@ func packageProjectConfigPath(cmd *cobra.Command, root string) string {
 	if cmd.Flags().Changed("project-config") {
 		return projectConfigPath
 	}
-	return filepath.Join(root, shop.DefaultConfigFileNameInDir(root))
+	return shop.SearchConfigPath(cmd.Context(), root, projectConfigPath)
```
Comment (MalteJanz) [inline cmd/project/project_deployment_package.go:24]: @shyim would be nice if you could take another look as I just resolved some merge conflicts from your PR: https://github.com/shopware/shopware-cli/pull/1539

In my commit: https://github.com/shopware/shopware-cli/pull/1432/changes/ef99b54d827b38a465ea1947525859316c08dcab..86737807f6378cc3c545fe13b4b8e8bd25521bb9

For example I removed this `DefaultConfigFileNameInDir` again as `SearchConfigPath` is the replacement I build. I didn't test my changes particularly for your deployment package command yet, but we could also do that in the demo if you join 🙂

File: internal/extension/config_init.go (hunk 3 above)
Comment (shyim) [reply internal/extension/config_init.go:21]: we only need to keep the extension config. the project one is never packaged and not needed really 🤔

Comment (MalteJanz) [reply internal/extension/config_init.go:21]: @shyim so you are saying the `.config/shopware-extension.yml` should be part of the extension ZIP and that is intentional? Then I think we should adjust the documentation 🙈

**Review by moshimorschi — APPROVED** [review]

File: internal/extension/config_init.go (hunk 3 above)
Comment (shyim) [reply internal/extension/config_init.go:21]: Yes that fine needs to be part of the zip. It should be already part 🤔

**Review by shyim — APPROVED** [review]

File: internal/extension/config_init.go (hunk 3 above)
Comment (MalteJanz) [reply internal/extension/config_init.go:21]: As discussed, I just confirmed the config will be part of the packaged extension as before and regardless of where the config is placed. So the docs were wrong and I removed that sentence here in [my docs PR](https://github.com/shopware/docs/pull/2521):
https://github.com/shopware/docs/pull/2521/changes/850061beca3291a14fcec33046976c79f60fc67f
